src/android/jni/fw_logger.cpp

```diff
@@ -2,27 +2,193 @@
 // Copyright(c) 2019 Intel Corporation. All Rights Reserved.
 
 #include <jni.h>
-#include <memory>
-
 #include "error.h"
-#include "../../../include/librealsense2/rs.h"
-#include "../fw-logger/rs-fw-logger.h"
+#include "../../../include/librealsense2/h/rs_internal.h"
+
+// Fw Logger methods
+
+extern "C"
+JNIEXPORT jlong JNICALL
+Java_com_intel_realsense_librealsense_FwLogger_nGetFwLog(JNIEnv *env, jclass clazz,
+                                                        jlong fw_logger_handle) {
+    rs2_error* e = NULL;
+    rs2_firmware_log_message* log_msg = rs2_create_fw_log_message(reinterpret_cast<rs2_device*>(fw_logger_handle), &e);
+    handle_error(env, e);
+
+    int result = rs2_get_fw_log(reinterpret_cast<rs2_device*>(fw_logger_handle), log_msg, &e);
+    handle_error(env, e);
+
+    return (jlong)log_msg;
+}
 
-std::shared_ptr<android_fw_logger> g_fw_logger;
-#define TAG "rs_fw_log"
+extern "C"
+JNIEXPORT jlong JNICALL
+Java_com_intel_realsense_librealsense_FwLogger_nGetFlashLog(JNIEnv *env, jclass clazz,
+                                                            jlong fw_logger_handle) {
+    rs2_error* e = NULL;
+    rs2_firmware_log_message* log_msg = rs2_create_fw_log_message(reinterpret_cast<rs2_device*>(fw_logger_handle), &e);
+    handle_error(env, e);
+
+    int result = rs2_get_flash_log(reinterpret_cast<rs2_device*>(fw_logger_handle), log_msg, &e);
+    if (result == 0){
+        e = rs2_create_error("No more logs in flash", "dummy_name", "dummy_args", RS2_EXCEPTION_TYPE_UNKNOWN);
```

Comment:
Why there is need for an error? Returning zero messages is not an exceptional UC.